libvast/src/system/catalog.cpp

```diff
@@ -516,27 +516,41 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
         num_partitions += 1;
         num_events += synopsis->events;
       }
+      auto total_num_partitions = count{0};
+      auto total_num_events = count{0};
       auto r = report{};
       r.data.reserve(num_partitions_and_events_per_schema_and_version.size());
       for (const auto& [schema_and_version, num_partitions_and_events] :
            num_partitions_and_events_per_schema_and_version) {
+        auto schema_num_partitions = num_partitions_and_events.first;
```

Comment:
Could be changed to one liner:
auto [num_partitions, num_events] = num_partitions_and_events;

But tbh it's not a big deal. Pick the one you like more :)

Reply:
Structured bindings are "more modern" and already used in the rest of the file - I applied your suggestion.